homeassistant/components/weatherkit/coordinator.py

```diff
@@ -62,10 +63,22 @@ async def _async_update_data(self):
             if not self.supported_data_sets:
                 await self.update_supported_data_sets()
 
-            return await self.client.get_weather_data(
+            updated_data = await self.client.get_weather_data(
                 self.config_entry.data[CONF_LATITUDE],
                 self.config_entry.data[CONF_LONGITUDE],
                 self.supported_data_sets,
             )
+
+            self.last_updated_at = datetime.now()
         except WeatherKitApiClientError as exception:
-            raise UpdateFailed(exception) from exception
+            if self.data is None or (
+                self.last_updated_at is not None
+                and datetime.now() - self.last_updated_at
+                > timedelta(seconds=STALE_DATA_THRESHOLD_SEC)
+            ):
+                raise UpdateFailed(exception) from exception
+
+            LOGGER.warning("Using stale data because update failed: %s", exception)
```

Comment:
It seems to happen on average 7 times per day based on my own logs. Do you think that’s too often for the warning level?

Reply:
It is expected behavior, so no need to "warn" about something that is totally normal.

Reply:
I wouldn't exactly call it "expected" since this is just a mitigation for a really weird issue (and thus users might want to be aware about it), but if problems arise then we can just ask people to enable debug logging. I agree that removing noise from logs is always a good idea, so I will change it to debug